rules/@typescript-no-watch/@typescript-no-watch.js

```diff
@@ -0,0 +1,64 @@
+const {
+  traverseNestedObjectNode,
+} = require("../../utils/traverse-nested-object-node");
+
+module.exports = {
+  meta: {
+    type: "suggestion",
+    docs: {
+      description: "Avoid `.watch` calls on any Effector unit or operator",
+      category: "Quality",
+      recommended: true,
+    },
+    messages: {
+      abusiveCall:
+        "Method `.watch` leads to imperative code. Try to replace it with operators (`sample`, `guard`, etc) or use the `target` parameter of the operators.",
+    },
+    schema: [],
+  },
+  create(context) {
+    const { parserServices } = context;
+
+    return {
+      CallExpression(node) {
+        const methodName = node.callee?.property?.name;
+
+        if (methodName !== "watch") {
+          return;
+        }
+
+        const object = traverseNestedObjectNode(node.callee?.object);
+        const objectName = object?.name;
+        const calleeName = object?.callee?.name;
+
+        if (!objectName && !["sample", "guard"].includes(calleeName)) {
```

Comment:
I do not understand the second part of check. I thought, we can check the return type of operator and apply rule. E.g. `sample` returns `Event`, so let's check it.

In my opinion, in this case, we can cover more cases — patronum-methods, for example.